src/odemis/driver/test/xt_client_test.py

```diff
@@ -286,36 +286,37 @@ def test_set_beam_shift(self):
     @unittest.skip("Currently this functionality is not supported by the XT client and the code is commented out")
     def test_apply_autostigmator(self):
         """
-        Test for the auto functionality of the apply_autostigmator.
-        Runs the apply_autostigmator multiple times and checks (using "get_stigmator()) if the found stigmator values
-        are equal each run. Checks the returned state, the return object and if an error is raised for an incorrect input state.
+        Test for the auto stigmation functionality.
         """
         detector = 'se-detector'  # Test autostigmator only for se-detector, channel=electron1
-        ## Basic testing of the functionality of the wrapper: can it be started and canceled
-        autostigmator_return = self.scanner.applyAutoStigmator(detector)
+        ## Start with basic check on starting and stopping the auto stigmation functionality
+        # Start auto stigmation and check it is running.
+        autostigmator_future = self.scanner.applyAutoStigmator(detector)
         time.sleep(2.5)  # Give microscope/simulator the time to update the state
         autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
         self.assertEqual(autostigmator_state, True)
-        self.assertIsInstance(autostigmator_return, ProgressiveFuture)
+        self.assertIsInstance(autostigmator_future, ProgressiveFuture)
 
-        autostigmator_return.cancel()
+        # Cancel auto stigmation process and check it is cancelled successfully.
+        autostigmator_future.cancel()
         time.sleep(5.0)  # Give microscope/simulator the time to update the state
         autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
         self.assertEqual(autostigmator_state, False)
 
-        ## Test if autostigmating gives repeatable results and check time it takes to auto stigmate.
+        ## Test if repeated autostigmating gives results within the allowed range and check the time it takes to auto
+        # stigmate.
 
-        # Create sets holding stigmator values to check if the outcome is consistent
-        stigmator_values_x = []  # List holding only the stigmator values in x direction
-        stigmator_values_y = []  # List holding only the stigmator values in y direction
+        # Create lists holding stigmator values to check if the outcome is within the allowed range.
+        stigmator_values_x = []  # List of stigmator values in x direction
+        stigmator_values_y = []  # List of stigmator values in y direction
 
-        # Run apply_autostigmator multiple times and afterwards check if the outcome is the same for each run
+        # Run apply_autostigmator multiple times and afterwards check if the outcome is within the allowed range
```

Comment:
I don't understand this test. I see in your code, that your run auto stigmation 5 times. You collect the values set on the stigmator HW after auto stigmation was finished. Then you check that these values are in the range of the allowed values for the stigmator HW. 
This should never be a problem I think. The values that you get from the stigmator are already set. So I would assume that if somehow a value is out of range, the autostigmation would already fail internally in xt and you will never reach the point to retrieve a value from the stigmator that is out of range.
Do I miss something?

Reply:
Actually this test isn't perfec, because the whole functionality is not implemented on the XT side.
Ideally we wanted to see if the function would give reproducible values from autostigmating which are more or less in the same range. But that test failed because the autostigmating didn't work. Thera suggested to at least check if the range information was given right and the correct values where returned. But as you say that is a really basic check which on second thought is not even needed to do. Note that currently the entire test cannot be runned because the autostigmation call doesn't work at all.
We can keep the list in there to allow easy implementation of a more elaborate test case when autostigmation works again. Do you think we should replace the assertLess/GreaterEqual statments with just a TODO?

Reply:
As discussed in the STUP:

- Either remove the check if "value set on stigmator is in range that is allowed by the stigmator HW" or if you leave it put a TODO to check if this can ever happen as you suggested.

- Add a TODO that running the autostigmation procedure should theoretically result in similar values set on the HW, but that the accuracy for this needs to be defined.